### ISSUE TITLE
This example has failed

### ISSUE THREAD

**Issue by zhengxingjian** [actix/actix#370]
```
 use actix::prelude::*;
 #[derive(Message)]
 #[rtype(result = "()")]
 struct Die;

 struct MyActor;

 impl Actor for MyActor {
     type Context = Context<Self>;
 }

 // To use actor with supervisor actor has to implement `Supervised` trait
 impl actix::Supervised for MyActor {
     fn restarting(&mut self, ctx: &mut Context<MyActor>) {
         println!("restarting");
     }
 }

 impl Handler<Die> for MyActor {
     type Result = ();

     fn handle(&mut self, _: Die, ctx: &mut Context<MyActor>) {
         ctx.stop();
 #       System::current().stop();
     }
 }

 fn main() {
     System::run(|| {
         let addr = actix::Supervisor::start(|_| MyActor);

         addr.do_send(Die);
     });
 }
```

The restarting method cannot be executed.

This example can be executed.

```
use actix::prelude::*;
use futures::FutureExt;

#[derive(Message)]
#[rtype(result = "i32")]
struct Die;
#[derive(Default)]
struct MyActor;
impl Actor for MyActor {
    type Context = Context<Self>;
}

// To use actor with supervisor actor has to implement `Supervised` trait
impl actix::Supervised for MyActor {
    fn restarting(&mut self, ctx: &mut Context<MyActor>) {
        println!("restarting");
    }
}
impl Handler<Die> for MyActor {
    type Result = ();

    fn handle(&mut self, _: Die, ctx: &mut Context<MyActor>) -> Self::Result {
        ctx.stop();
        println!("Die");
        //       System::current().stop();
    }
}

fn main() {
    let system = System::new("test");

    let addr = actix::Supervisor::start(|_| MyActor);

    addr.do_send(Die);
    // send message and get future for result
    println!("second");
    system.run();
}
```

**Comment by Jonathas-Conceicao** [comment]
The problem with the first example is that the system will only actually start to run _after_ the provided closure as executed, and by the time the closure finished the `Addr` will be dropped, and with no live references the `Actor` will stop. If we make the `Addr` live logger than the closure you will see the "restarting" being printed to stdout.

Here a quick way to make the `Addr` not being dropped on the first example:
```Rust
fn main() {
    System::run(|| {
        let addr = actix::Supervisor::start(|_| MyActor);

        addr.do_send(Die);
        std::mem::forget(addr);
        println!("second");
    });
}
```

The reason this works as you expect in the second example is that when `system.run()` is called the `Addr` is still in scope.

**Comment by JohnTitor** [comment]
I'm going to close this issue in favor of https://github.com/actix/actix/issues/370#issuecomment-603619669, thanks!

**Comment by rubberduck203** [comment]
Maybe this should  be reopened to track the documentation update.
As of 0.10.0-alpha.3 the example in the doc is still broken.

https://docs.rs/actix/0.10.0-alpha.3/actix/struct.Supervisor.html

**Comment by robjtede** [comment]
Included a fix for this in the v0.10 release PR. Not using `mem::forget`, seems like something to avoid in an example but this is largely a guess because I'm not too familiar with this part of actix.
